extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -397,34 +393,52 @@ public function getArticleFromMarkup() {
 		$this->response->setVal( 'wikiVariables', $wikiVariables );
 	}
 
-	/**
-	 * @throws NotFoundApiException
-	 * @throws BadRequestApiException
-	 */
-	public function getArticle() {
+	public function getPage() {
 		global $wgEnableMainPageDataMercuryApi, $wgCityId;
 
 		try {
 			$title = $this->getTitleFromRequest();
 			$articleId = $title->getArticleId();
-			$sections = $this->getVal( 'sections' );
 
-			// getArticle is cached (see the bottom of the method body) so there is no need for additional caching here
+			// getPage is cached (see the bottom of the method body) so there is no need for additional caching here
 			$article = Article::newFromID( $articleId );
 
-			if ( $title->isRedirect() ) {
+			if ( $article instanceof Article && $title->isRedirect() ) {
 				/* @var Title|null $redirectTargetTitle */
 				$redirectTargetTitle = $article->getRedirectTarget();
-				$data[ 'redirected' ] = true;
-				if ( $redirectTargetTitle instanceof Title && !empty( $redirectTargetTitle->getArticleID() ) ) {
-					$article = Article::newFromID( $redirectTargetTitle->getArticleID() );
+				$redirectTargetID = $redirectTargetTitle->getArticleID();
+				$data['redirected'] = true;
+
+				if ( $redirectTargetTitle instanceof Title && !empty( $redirectTargetID ) ) {
 					$title = $redirectTargetTitle;
+					$articleId = $redirectTargetID;
+					$article = Article::newFromID( $redirectTargetID );
 				} else {
-					$data[ 'redirectEmptyTarget' ] = true;
+					$data['redirectEmptyTarget'] = true;
 				}
 			}
 
-			if ( !$article instanceof Article ) {
+			$isMainPage = $title->isMainPage();
+			$data['isMainPage'] = $isMainPage;
+
+			$titleBuilder = new WikiaHtmlTitle();
+
+			if ( $isMainPage &&
+				!empty( $wgEnableMainPageDataMercuryApi ) &&
+				( new CommunityDataService( $wgCityId ) )->hasData()
+			) {
+				$data['mainPageData'] = $this->getMainPageData();
+			} elseif ( $title->isContentPage() ) {
```

Comment:
what about checking here if `$article instanceof Article` ? if no, further processing doesn't make sense because `getArticleData` will be [] and nothing will happen